### ISSUE TITLE
[many ports] Dependency on vcpkg-cmake-config but not using vcpkg_cmake_config_fixup

### ISSUE THREAD

**Issue by Thomas1664** [microsoft/vcpkg#24430]
This seems to be a common copy and paste error.
I used a Python script to find out if there are any ports that have a dependence on `vcpkg-cmake-config` but don't use `vcpkg_cmake_config_fixup`. The interesting lines are:

```python
if content.find("vcpkg-cmake-config") == -1 or content.find("dependencies") == -1:
          continue
[...]
if content.find("vcpkg_cmake_config_fixup") == -1:
          print(f"- [ ] {entry.name}")
```

Maybe there is a possibility to add this as a PR suggestion/ manifest format/ post-build check?

- [ ] arrayfire
- [ ] bext-wintls
- [ ] c89stringutils
- [ ] clickhouse-cpp
- [ ] clipboardxx
- [ ] cpp-redis
- [ ] cppad
- [ ] cppunit
- [ ] cspice
- [ ] distorm
- [ ] easyloggingpp
- [ ] eve
- [ ] fastor
- [ ] graphqlparser
- [ ] gsl
- [ ] hypodermic
- [ ] ijg-libjpeg
- [ ] jack2
- [ ] jasper
- [ ] klein
- [ ] kwsys
- [ ] laszip
- [ ] launch-darkly-server
- [ ] libcurl-simple-https
- [ ] liblinear
- [ ] libopensp
- [ ] libressl
- [ ] libwebm
- [ ] lionkor-commandline
- [ ] live555
- [ ] lmdb
- [ ] lunasvg
- [ ] mbedtls
- [ ] minio-cpp
- [ ] mygui
- [ ] nanobench
- [ ] opencensus-cpp
- [x] openssl
- [ ] protobuf
- [ ] protozero
- [ ] ptyqt
- [ ] randomstr
- [ ] redis-plus-plus
- [ ] rhasheq
- [ ] sbp
- [ ] skia
- [ ] sockpp
- [ ] spirv-reflect
- [ ] tmxlite
- [ ] unicorn-lib
- [ ] usockets
- [ ] winpty
- [ ] wxwidgets
- [ ] yasm
- [ ] zziplib
- [ ] pcre2

**Comment by Neumann-A** [comment]
qtbase is a false positiv.

**Comment by dg0yt** [comment]
IMO `vcpkg_cmake_config_fixup` should be integrated, maybe via a dependency,  into vcpkg_cmake.
Is there a stand-alone usage?

**Comment by Thomas1664** [comment]
> IMO `vcpkg_cmake_config_fixup` should be integrated, maybe via a dependency, into `vcpkg-cmake`. Is there a stand-alone usage?

Yes, these ports use `vcpkg_cmake_config_fixup` independently from `vcpkg_cmake_*`:
- gpgme
- libgwenhywfar
- libsmb2
- tomlplusplus

**Comment by dg0yt** [comment]
I still think that `vcpkg_cmake_config_fixup` should be integrated into `vcpkg-cmake`.

**Comment by Thomas1664** [comment]
no stale

**Comment by Thomas1664** [comment]
no stale
